components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java

```diff
@@ -102,8 +122,20 @@ protected void initializeAuthenticator() {
 
     protected Authenticator getAuthenticator() {
         if (authenticator == null) {
-            authenticator = new OAuthAuthenticator();
+            if (authorizationHeader == null) {
```

Comment:
Can authorizationHeader be empty(authorizationHeader = "")? if so we may have to check for that as well.

Reply:
There will not be a scenario where the value will be set as "" because that is handled when the synapse file is generated when publishing an API. This will either be "null" or some actual string with characters.
Should I made the changes to check "" anyway ?